model/host/host.go

```diff
@@ -3370,3 +3475,233 @@ func (h *Host) UnsetPersistentDNSInfo(ctx context.Context) error {
 
 	return nil
 }
+
+// SleepScheduleSentinelTime is a special date that's so far in the future that
+// it's assumed Evergreen will never reach this time. This timestamp is only
+// used to signify that a host's sleep schedule should not take effect (either
+// is permanently exempt from the sleep schedule, or is being kept off
+// indefinitely).
+var SleepScheduleSentinelTime = time.Date(10000, 0, 0, 0, 0, 0, 0, time.UTC)
+
+// GetNextScheduledStopTime returns the next time a host should be
+// stopped according to its sleep schedule.
+func (h *Host) GetNextScheduledStopTime(now time.Time) (time.Time, error) {
+	if h.SleepSchedule.PermanentlyExempt || h.SleepSchedule.ShouldKeepOff {
+		return SleepScheduleSentinelTime, nil
+	}
+
+	if err := h.SleepSchedule.Validate(); err != nil {
+		return time.Time{}, errors.Wrap(err, "invalid sleep schedule")
+	}
+
+	userTimeZone, err := time.LoadLocation(h.SleepSchedule.TimeZone)
```

Comment:
We could consider putting a `*time.Location` on the  sleep schedule info itself that gets populated during validate because validate is already running and `LoadLocation` and returning an error if it fails